guides/micronaut-dependency-injection-types/micronaut-dependency-injection-types.adoc

```diff
@@ -0,0 +1,78 @@
+common:header-top.adoc[]
+
+== Sample Project
+
+You can link:@sourceDir@.zip[download a sample application] with the code examples shown in this article.
+
+== Introduction
+
+What is a dependency? Class A has a dependency on Class B when it interacts with it in any way. For example, Class A executes a method in an instance of Class B. You typically don't instantiate class dependencies when using Micronaut Framework's dependency injection engine. Instead, you rely on the framework to provide dependency instances.
+
+NOTE: In Micronaut Framework documentation, we often use the term `Bean` and `Dependency` interchangeably.
+
+== JSR-330
+
+Micronaut Framework implements the https://javax-inject.github.io/javax-inject/[JSR-330 - Dependency Injection for Java specification].
+ 
+== jakarta.inject vs javax.inject
+ 
+Early versions of Micronaut Framework used `javax.inject` annotations. Due to trademark restrictions imposed on the `javax.*` namespace, the Micronaut framework switched from the `javax.inject` to the `jakarta.inject` annotations. Since version https://micronaut.io/2021/03/09/micronaut-2-4-released/[Micronaut Framework 2.4] supports `jakarta.inject` annotations. Moreover, Micronaut Framework switched to `jakarta.inject` as the set of annotations included by default with the Framework, and we strongly encourage you to use `jakarta.inject` moving forward.
```

Comment:
```suggestion
Early versions of Micronaut Framework used `javax.inject` annotations. Due to trademark restrictions imposed on the `javax.*` namespace, the Micronaut framework switched from the `javax.inject` to the `jakarta.inject` annotations. Since https://micronaut.io/2021/03/09/micronaut-2-4-released/[version 2.4], the Micronaut framework supports `jakarta.inject` annotations. Moreover, Micronaut Framework switched to `jakarta.inject` as the set of annotations included by default, and we strongly encourage you to use `jakarta.inject` going forward.
```